[#15225] REST invocation helper post start

jabolina opened the pull request:
* Initialize InvocationHelper dependencies in the postStart phase.
Fixes #15225.

wburns commented:
@jabolina Can you confirm test hang is not related?

karesti commented on server/tests/src/test/java/org/infinispan/server/functional/rest/RestProtobufResource.java:
```diff
@@ -0,0 +1,42 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+public class RestProtobufResource {
```
@jabolina could we rename this file, please ? makes think is an actual class instead of a test

wburns replied:
:+1: to naming it *Test as we are doing that actually https://github.com/infinispan/infinispan/pull/15231/files#diff-b7bb2105a3c3bfe5ab29dd75bb03f4c925bb7cdc5dd3d1ad5b69190701ff8224R73

tristantarrant replied:
I've checked: the `server/tests` pom is disabling surefire, we should be able to safely use `*Test`, although it will only be executed as part of a parent `IT` test from failsafe

karesti reviewed (changes requested):
+1 @wburns question, besides my minor comment, LGTM

jabolina commented:
> @jabolina Can you confirm test hang is not related?

I've pushed a new commit. From the logs that GH shows, the agent hanged while running the core tests.

tristantarrant commented:
Coverage dropped below 75%

jabolina commented:
I've renamed the new test and few more changes. Included a new test in the REST module that invokes the endpoint during different stages, and I found the issue could still happen. I've added a new option to building the `Invocation` to notify whether the endpoint requires the cache manager to be running. If the endpoint requires the cache manager, and we know it still haven't fully initialized, we automatically return 503. This is similar to what we have in the other protocols (Hot Rod, RESP, and Memcached).

jabolina commented:
And it also fixes #15221 because the endpoints now return 503 until everything is initialized.
